src/action.js

```diff
@@ -31,34 +32,60 @@ module.exports = function link(config, args) {
   }
 
   const packageName = args[0];
-  const dependencies = packageName ? [packageName] : getProjectDependencies();
 
-  dependencies
-    .forEach(name => {
-      const dependencyConfig = config.getDependencyConfig(name);
-
-      if (!dependencyConfig) {
-        return log.warn('ERRINVALIDPROJ', `Project ${name} is not a react-native library`);
+  const dependencies =
+    (packageName ? [packageName] : getProjectDependencies())
+    .map(name => ({
+      config: config.getDependencyConfig(name),
+      name,
+    }))
+    .filter(dependency => {
+      if (!dependency.config) {
+        log.warn(
+          'ERRINVALIDPROJ',
+          `Project ${dependency.name} is not a react-native library`
+        );
+        return false;
       }
+      return true;
```

Comment:
That whole block from 36 to 50 is potentially a good candidate for separate method so we can clean up the main function. But I'd prefer to wait for the unit tests and make sure it actually works w/o any issues (from my internal testing all's been good, but you know)